src/client/Client.cc

```diff
@@ -4388,10 +4388,12 @@ void Client::trim_caps(MetaSession *s, uint64_t max)
 	  all = false;
         }
       }
+      if (in->ll_ref == 1 && in->ino != MDS_INO_ROOT) {
+         _schedule_ino_release_callback(in.get());
+      }
```

Comment:
Ok, I get the point about this entry being lru_pinned, but why the test for ll_ref == 1? Shouldn't we issue a request to release the callback no matter what the inode's refcount is?

Reply:
Yes，we can issue the request to release callback no matter what the inode's ref is, if inode's ll_ref not equal to 1,  Client can not release cache even though the ino_release_callback function is called. At this time, i think it's better to keep cache in use client like nfs-ganesha. Right?

Reply:
Yes, probably so. Typically ganesha will only keep a single reference for its cached filehandle, but it could (e.g.) find the inode again during a lookup or something and take an extra reference to it. In that case though, that's probably an indicator that the thing is still in use and shouldn't be purged.

Reply:
As far as I am aware, ganesha only keep a single reference(ll_ref) for inode when object_handle can be released in nfs-ganesha mdcache,  because ganesha itself has a cache reference count.
Should I remove "in->ll_ref == 1"(we can sure that whatever the in->ll_ref is, we can call the release_callback  to issue request)?  I'm not sure if there are any extra references.